File: tests/transpiler/x/pas/tail_recursion.pas
{$mode objfpc}
program Main;
function sum_rec(n: integer; acc: integer): integer;
begin
  if n = 0 then begin
  exit(acc);
end;
  exit(sum_rec(n - 1, acc + n));
end;
begin
  writeln(sum_rec(10, 0));
end.

File: tests/transpiler/x/fortran/tail_recursion.f90
program main
  implicit none
  print '(I0)', sum_rec(10, 0)
contains
  function sum_rec(n, acc) result(res)
    implicit none
    integer :: res
    integer :: n
    integer :: acc
    if (n == 0) then
      res = acc
      return
    end if
    res = sum_rec(n - 1, acc + n)
    return
  end function sum_rec
end program main
